pkg/types/installconfig.go

```diff
@@ -119,6 +119,12 @@ type InstallConfig struct {
 	// +kubebuilder:default=false
 	// +optional
 	FIPS bool `json:"fips,omitempty"`
+
+	// ForceCredentialsMode instructs the installer to not attempt to query the cloud permissions before attempting
+	// installation. It also passes down the desired credentials mode to the cloud-credential-operator so that it too
+	// does not attempt to query permissions.
+	// +optional
+	ForceCloudCredentialsMode CloudCredentialsMode `json:"forceCloudCredentialsMode,omitempty"`
```

Comment:
```suggestion
	CloudCredentialsMode CloudCredentialsMode `json:"cloudCredentialsMode,omitempty"`
```

Reply:
Force is just not a good api imo, `cloudCredentialsMode` is clear enough for users. and we can document that installer and cred minter will detect the best mode in case it's empty

Reply:
Secondly I think we should add a validation that allows users to set this field only for certain platforms, it doesn't make sense to allow users to set this field for vSphere, oVirt, None etc.

Reply:
I will add validation.